### PR TITLE
fix: Add support for forced tunneling

### LINKED ISSUE #3368: [AVM Module Issue]: Unable to create management IP using standard SKU in `avm/res/network/azure-firewall`
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Module Name

avm/res/network/azure-firewall

### (Optional) Module Version

latest

### Description

I tried to use the module `avm/res/network/azure-firewall` to setup an Azure Firewall including a management IP and `AzureFirewallManagementSubnet` so I could get forced tunneling up and running. I didn't get it to work using a `Standard` SKU, and looking at the module in the current state it does not support this using a SKU other than `Basic`.

I would like to implement support for Standard and Premium SKU, so I went with this approach:
* Introduced a parameter called `forcedTunneling` with the `bool` type with standard value `false`
* Updated `requiresManagementIp` so it checks the following:

```bicep
var requiresManagementIp = azureSkuTier == 'Basic' || azureSkuTier == 'Standard' && forcedTunneling || azureSkuTier == 'Premium' && forcedTunneling ? true : false
```

The above changes worked for me and allowed me to deploy an Azure Firewall with `Standard` SKU and it created the management IP. Am I missing something, or would you like to see another approach @hundredacres ?

### (Optional) Correlation Id

_No response_

### PR DESCRIPTION
**Pull request by hundredacres**
## Description
Adds param to enable forcedTunneling, which adds a managementIPAddress.

Fixes #3368 
Closes #3368 

## Pipeline Reference

| Pipeline |
| -------- |
[avm.res.network.azure-firewall](https://github.com/hundredacres/bicep-registry-modules/actions/workflows/avm.res.network.azure-firewall.yml)       

## Type of Change

- [ ] Update to CI Environment or utilities (Non-module affecting changes)
- [X] Azure Verified Module updates:
  - [X] Bugfix containing backwards-compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [X] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

## Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change
- [X] I have run `Set-AVMModule` locally to generate the supporting module files.
- [X] My corresponding pipelines / checks run clean and green without any errors or warnings

### CODE REVIEW

**Comment by ReneHezser** [comment]
@hundredacres 
We have added examples for Bicep parameter files to the Readme. This has been applied to all published modules but needs to be done for PRs as well. Can you please run the Set-AVMModule utility as detailed [here](https://azure.github.io/Azure-Verified-Modules/contributing/bicep/bicep-contribution-flow/generate-bicep-module-files/). It is required for the validation pipeline to succeed and the contribution to be published.

Please reach out if any support is needed.

**Comment by hundredacres** [comment]
@ReneHezser / @Azure/avm-res-network-azurefirewall-module-owners-bicep can I get a review here?

**Comment by AlexanderSehr** [comment]
> @ReneHezser / @Azure/avm-res-network-azurefirewall-module-owners-bicep can I get a review here?

Hey @hundredacres, sorry my bad. I saw that @rahalan is part of the owner team and hence assumed he's a co-owner who'd do the review. After checking the AVM page, I can see that's not the case. I'd recommend to probably 'fix' that to avoid the confusion in the future 😉

File: avm/res/network/azure-firewall/tests/e2e/tunneling/main.test.bicep [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      enableForcedTunneling: true
+      managementIPAddressObject: {
+        publicIPAllocationMethod: 'Static'
+        roleAssignments: [
+          {
+            roleDefinitionIdOrName: 'Reader'
+            principalId: nestedDependencies.outputs.managedIdentityPrincipalId
+            principalType: 'ServicePrincipal'
+          }
+        ]
```
Comment (AlexanderSehr) [inline avm/res/network/azure-firewall/tests/e2e/tunneling/main.test.bicep:74]: Why the role assignment? I'd suggest to keep the test focused on the features you want to test to not accidently imply that e.g. a role assignment would need to be done in order to deploy forced tunneling

Comment (hundredacres) [reply avm/res/network/azure-firewall/tests/e2e/tunneling/main.test.bicep]: Updated.

**Review by AlexanderSehr — APPROVED** [review]

**Comment by AlexanderSehr** [comment]
Great work @hundredacres, off we go 🌊

**Pull request merged**